### PR TITLE
Document how to create a Session

### PR DESCRIPTION
**Pull request by andybalaam**
On a recent [live stream](https://diode.zone/w/uSPoZBCu6zs2fNmsotHRiK?start=38m15s) I spent ages trying to create a `matrix_sdk::Session`.

I think I worked out the right way to make one, but if I did it wrong, please let me know and I'll update the PR.

### CODE REVIEW

File: crates/matrix-sdk-base/src/session.rs [hunk 1]
```diff
@@ -20,6 +20,22 @@ use serde::{Deserialize, Serialize};
...
+///
+/// # Example
+///
+/// ```
+/// use matrix_sdk_base::Session;
+/// use ruma::UserId;
+/// use std::str::FromStr;
+///
+/// let session = Session {
+///     access_token: String::from("abcdefgh012345678"),
```
Comment (poljar) [inline crates/matrix-sdk-base/src/session.rs:32]: Let's copy the example from `Client::restore_login()` instead, that one doesn't have the unwraps and I think `from_str()` is quite uncommon.

If you think we should showcase how to dynamically create the user ID instead, this one would be preferable:

```
/// use matrix_sdk_base::Session;
/// use ruma::UserId;
///
/// # fn main() -> anyhow::Result<()> {
/// let session = Session {
///     access_token: String::from("My-Token"),
///     user_id: UserId::try_from("@example:localhost")?,
///     device_id: "DEVICEID".into()
/// };
/// # Ok(())
/// # }
```

Don't forget to add `anyhow = "1.0.45"` to the `Cargo.toml` file of the base crate.

**Review by poljar — CHANGES_REQUESTED** [review]

File: crates/matrix-sdk-base/src/session.rs (hunk 1 above)
Comment (andybalaam) [reply crates/matrix-sdk-base/src/session.rs]: Thanks, I added a commit based on the `Client::restore_login()` example.

**Comment by jplatte** [comment]
Maybe a `Session::from(login::Response)` impl should be provided? (or a dedicated method, for better discoverability)

**Comment by poljar** [comment]
The `anyhow` wasn't needed in the first variant since `user_id!()` statically makes sure that the ID is valid, the example didn't need any `?` to throw errors.

I modified this myself. Thanks, merged.

**Comment by poljar** [comment]
> Maybe a `Session::from(login::Response)` impl should be provided? (or a dedicated method, for better discoverability)

This will be added in #418.

**Comment by andybalaam** [comment]
Fantastic, thank you!

**Pull request merged**